### DISCUSSION TITLE
I have a question about the structure of addUtilities plugin

### DISCUSSION THREAD

**Discussion by youngseo-im** [tailwindlabs/tailwindcss#13910] [Help] [answered]
What I want to do is to use only the 'text-primary' class to support dark mode

As shown in the attached image, '@apply text-gray-900 dark:text-gray-50' is a key, not a value, why is it structured this way?

The expected value is
` '.text-primary' : '@apply text-white dark:text-gray-50'`

And I'm wondering if there is a better way to fulfill my requirements : )

![image](https://github.com/tailwindlabs/tailwindcss/assets/75972718/056b925e-4f96-4157-9c5a-10acfdf65f4b)

![image](https://github.com/tailwindlabs/tailwindcss/assets/75972718/813f2bed-6c3c-4253-b03f-e75ced2bbac4)

**Comment by wongjn** [comment]
> `@apply text-gray-900 dark:text-gray-50` is a key, not a value, why is it structured this way?

It is how the underlying PostCSS parser expects at-rules to be structured.

> And I'm wondering if there is a better way to fulfill my requirements : )

Generally, the maintainers don't really recommend this approach and thus is probably why there is not a more "built-in" way of achieving this.

For example, from [Simon Vrachliotis](https://github.com/tailwindlabs/tailwindcss/discussions/3583#discussioncomment-368696):
> When implementing dark mode interfaces, you likely don't always want to just "swap" or invert colors, and often need much more "fine control", to make contextual decisions on a lot of parts of the UI.

And from [Adam Wathan](https://github.com/tailwindlabs/tailwindcss/discussions/10274#discussioncomment-4627634):
> Personally I think doing dark mode this way is seductive but not as practical as you expect. Tons of situations where you end up swapping the same color in light mode for two different colors in dark mode depending on how it’s used, and then you need to come up with different names for those two colors and it quickly becomes a very big project.

You could consider using the classes directly on the elements instead.

**Reply by youngseo-im** [reply]
@wongjn 

Isn't it recommended to use apply?

If so, is it more advisable to add the class individually?

**Reply by wongjn** [reply, marked as answer]
It is not recommended to use `@apply` – Adam Wathan (creator of Tailwind) does seem to advocate avoiding it:
- https://twitter.com/adamwathan/status/1226511611592085504
- https://twitter.com/adamwathan/status/1559250403547652097
